grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingService.java

```diff
@@ -588,6 +613,20 @@ private HttpResponse serve0(ServiceRequestContext ctx, HttpRequest req,
         return HttpResponse.of(responseFuture);
     }
 
+    private HttpData convertToArbitraryHttpData(ServiceRequestContext ctx,
+                                                AggregatedHttpRequest request) throws IOException {
+        final ObjectNode body = mapper.createObjectNode();
+
+        try {
+            body.put("content_type",  request.contentUtf8());
+            body.put("data", request.content().array());
```

Comment:
Out of curiosity, how is a byte array serialized into JSON? How do we deal with a non-UTF8 data?

Reply:
Let me write some tests, in protobuf spec it should be base64 array.

Reply:
Looking back to this, I have realized this is still WIP. I'll come back.

I'm still trying to comprehend how transcoding works internally 🙂.

Reply:
Sure. Please take your time. I appreciate that you continue working on this :heart_decoration:

Reply:
I have tested it out and turns out it uses base64. So in that case whether it is UTF8 or not should not matter, right?

Reply:
So rest of the code should be self-explanatory. We will see.